### PR TITLE
feat: add debug emotes for testing

### PR DESCRIPTION
**Pull request by fcolarich**
## What does this PR change?

I piggybacked on a change made by Nico to load emotes from command args, so now we can also load emotes from the debug settings. This allows to test specific emotes more easily from the editor.

## Summary by CodeRabbit

- **New Features**
	- Enhanced user profiles with the ability to dynamically add emotes based on debug settings.
	- Improved loading spinner visuals for better user feedback during loading processes.

- **Bug Fixes**
	- Ensured that the `forcedEmotes` list is always initialized, reducing potential null reference issues.

- **Documentation**
	- Updated interface and class definitions to reflect new properties and methods for handling emotes.

### CODE REVIEW

File: Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs [hunk 1]
```diff
@@ -260,7 +260,7 @@ IMultiPool MultiPoolFactory() =>
             var equippedEmotes = new EquippedEmotes();
             var forceRender = new List<string>();
             var selfProfile = new SelfProfile(container.ProfileRepository, identityCache, equippedWearables, wearableCatalog,
-                emotesCache, equippedEmotes, forceRender, ParseSelfForcedEmotes(bootstrapContainer.ApplicationParametersParser));
+                emotesCache, equippedEmotes, forceRender, ParseSelfForcedEmotes(bootstrapContainer.ApplicationParametersParser), ParseDebugForcedEmotes(bootstrapContainer.DebugSettings.EmotesToAddToUserProfile));
```
Comment (lorux0) [inline Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs:263]: We could inject a single list by merging both, the ones from debug settings and the app params, with no need to modify the `SelfProfile` class. What do you think?

**Review by lorux0 — CHANGES_REQUESTED** [review]

File: Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs (hunk 1 above)
Comment (fcolarich) [reply Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs]: Sounds good!

File: Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs [hunk 2]
```diff
@@ -634,15 +634,21 @@ IMultiPool MultiPoolFactory() =>
             return (container, true);
         }
 
-        private static URN[]? ParseSelfForcedEmotes(IAppArgs appParams)
+        private static IReadOnlyList<URN>? ParseForcedEmotes(IAppArgs appParams, IReadOnlyList<string>? debugEmotes)
```
Comment (lorux0) [inline Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs:637]: I think its wrong to add the responsibility to this method. It should only do one thing, parse the urns and retrieve the list. If you want to add the debug emotes, then you should do it where its called. Something like:
```
List<URN> selfEmotes = new List<URN();

IReadOnlyList<URN>? emotesThroughParams = ParseForcedEmotes(bootstrapContainer.ApplicationParametersParser);

if (emotesThroughParams != null)
    selfEmotes.AddRange(emotesThroughParams);
    
if (bootstrapContainer.DebugSettings.EmotesToAddToUserProfile != null)
    selfEmotes.AddRange(bootstrapContainer.DebugSettings.EmotesToAddToUserProfile);
    
var selfProfile = new SelfProfile(container.ProfileRepository, identityCache, equippedWearables, wearableCatalog,
                emotesCache, equippedEmotes, forceRender, selfEmotes);
....
```

**Review by lorux0 — APPROVED** [review]

File: Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs (hunk 2 above)
Comment (fcolarich) [reply Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs:637]: your proposal sounds cleaner! nice 👍

**Review by davidejensen — APPROVED** [review]

**Pull request merged**